pkg/cluster/kubeclient/backup.go

```diff
@@ -0,0 +1,94 @@
+package kubeclient
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/openshift-azure/pkg/util/randomstring"
+	"github.com/openshift/openshift-azure/pkg/util/ready"
+)
+
+func (u *kubeclient) BackupCluster(ctx context.Context, backupName string) (string, error) {
+	if backupName == "" {
+		backupName, err := randomstring.RandomString("abcdefghijklmnopqrstuvwxyz0123456789", 5)
+		if err != nil {
+			return "", fmt.Errorf("could not generaate backup name")
+		}
+		backupName = "etcd-backup-" + backupName
+	}
+	backupImage := "quay.io/openshift-on-azure/etcdbackup:latest"
+	// Enable backup code PR testing with the ETCDBACKUP_IMAGE variable.
+	if os.Getenv("ETCDBACKUP_IMAGE") != "" {
+		backupImage = os.Getenv("ETCDBACKUP_IMAGE")
+	}
+
+	u.log.Infof("running an etcd backup using %s", backupImage)
+	bk, err := u.client.BatchV1().Jobs("openshift-etcd").Create(&batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      backupName,
+			Namespace: "openshift-etcd",
+		},
+		Spec: batchv1.JobSpec{
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					NodeSelector:       map[string]string{"node-role.kubernetes.io/master": "true"},
+					ServiceAccountName: "etcd-backup",
+					RestartPolicy:      "Never",
+					Containers: []v1.Container{
+						{
+							Name:            "etcdbackup",
+							Image:           backupImage,
+							ImagePullPolicy: "Always",
+							Args:            []string{fmt.Sprintf("-blobname=%s", backupName), "save"},
+							VolumeMounts: []v1.VolumeMount{
+								{
+									Name:      "azureconfig",
+									MountPath: "/_data/_out",
+									ReadOnly:  true,
+								},
+								{
+									Name:      "origin-master",
+									MountPath: "/etc/origin/master",
+									ReadOnly:  true,
+								},
+							},
+						},
+					},
+					Volumes: []v1.Volume{
+						{
+							Name: "azureconfig",
+							VolumeSource: v1.VolumeSource{
+								HostPath: &v1.HostPathVolumeSource{
+									Path: "/etc/origin/cloudprovider",
+								},
+							},
+						},
+						{
+							Name: "origin-master",
+							VolumeSource: v1.VolumeSource{
+								HostPath: &v1.HostPathVolumeSource{
+									Path: "/etc/origin/master",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	})
+	if err != nil {
+		return "", err
+	}
+	err = wait.Poll(2*time.Second, 5*time.Minute, ready.BatchIsReady(u.client.BatchV1().Jobs(bk.Namespace), bk.Name))
+	if err != nil {
+		return "", err
+	}
+	return backupName, nil
```

Comment:
should check if the job was successful and return accordingly
should delete the job